src/iceberg/transform_function.h

```diff
@@ -97,11 +97,11 @@ class YearTransform : public TransformFunction {
   /// \param source_type Must be a timestamp type.
   explicit YearTransform(std::shared_ptr<Type> const& source_type);
 
-  /// \brief Extracts the year from each timestamp in the input array.
-  Result<ArrowArray> Transform(const ArrowArray& input) override;
+  /// \brief Extract a date or timestamp year, as years from 1970.
```

Comment:
The comment is incorrect.

Reply:
Also uses `Extracts` to be consistent. Same for below.

Reply:
I agree the comment is not clear, but that's from the spec, see [1]. We might also want to change the spec if it's not clear.

[1] https://iceberg.apache.org/spec/#partition-transforms

Reply:
iceberg-rust use the same description[1], so I'm inclined to keep it consistent.

[1]: https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/transform/temporal.rs#L44